Dockerfile

```diff
@@ -0,0 +1,44 @@
+ARG NODE_VERSION="12.2.0"
+FROM library/node:${NODE_VERSION} AS development
+
+ENV NODE_ENV "development"
+WORKDIR /usr/src/freshlytics
+
+COPY package*.json ./
+RUN npm install --no-optional
+
+COPY . .
+
+CMD ["npm", "run", "dev"]
+EXPOSE 3000/tcp 3001/tcp
+
+FROM development AS builder
+
+ENV NODE_ENV "production"
+RUN npm run build
```

Comment:
I'm a bit confused about this part. Would this stage retain the node_modules from the previous stage?

Reply:
Yup, this part retains the node_modules from the previous build stage (development). The reason I do it this way is so that you have less steps to run (since the beginning of the build stage would be the exact same as the development stage), and you're able to use the development stage as a target in the docker-compose file.

Reply:
I understood the reason for my confusion - we're using the same image as previous stage (`FROM development AS builder`) here so that's probably why we can reuse the node_modules 👍Thanks for explaining the reason!